### ISSUE TITLE
Multipage Canvas POC

### ISSUE THREAD

**Issue by miina** [swissspidy/web-stories-wp#9643]
## Feature Description

Edit: See https://github.com/GoogleForCreators/web-stories-wp/issues/9643#issuecomment-1064304530 for the latest update. Design TBD.

Important: UX prefers the feature to be released once it's fully ready, there won't be originally considered MVP etc. even though we'll most likely implement it in multiple steps. Once we know better, we can convert this issue into an epic.

See [Figma](https://www.figma.com/file/dc9xrVlRhQq4KTE9WQzU1R/_Editor---Sprint-Feature-Level-Designs_?node-id=10548%3A453241)

Note: this feature has been blocked for some time due to potential hit on performance. It has been brought up again by UX as high priority so we should take another look into this to find a way how to implement it.

Some options for unblocking this:
- Improve the Page performance (we should do this anyway, also see #8255). EDIT: Note that various performance improvements have been done by Pea pod meanwhile, some of those also influence the Page rendering time.
- Perhaps we can use the images generated anyway for the page carousel thumbnails. The images are tiny though, and when loading the first time, these are not generated yet, so this needs some looking into. EDIT: See #9749 

<img src="https://user-images.githubusercontent.com/637548/173100843-3114a49a-c01e-428c-b42d-f3d7f549dc57.png" />

**Comment by miina** [comment]
UX brought this issue up as quite important and requested.
We should take another look at this, most likely the issue will be done in two steps:
- Displaying non-interactive pages (able to navigate by clicking but not able to drag anything to the next page etc)
- Displaying partially interactive pages: possible to drag an element from the active page to prev-next pages.

First we need to fix the thumbnail generation to only generate one image per page, we could potentially reuse those for the prev/next pages.

**Comment by miina** [comment]
Update from design sync on the 9th of March

The ideal option from Amy:

Make the entire story interactive and side-to-side scrollable

Implications:
- Users will be able to see how objects line up from canvas to canvas, and potentially have point guides in place for exactness
- Users can copy, paste, and drag objects between all canvases
- Users can edit any slide anywhere on screen by scrolling to it and putting it wherever they feel comfortable editing it, even if partial view
- Scrolling would work somewhat like Canva
- There will be no “main canvas” indicators in the workspace, but perhaps an indicator of the canvas you’re working on at any given time on click or nav (for accessibility?)
- Each canvas would have a canvas menu, but quick menu doesn’t show up till you click on it

**Comment by bmattb** [comment]
@aaskedall Can you confirm that this is still desired? Prom is thinking about picking this up in next sprint. cc @swissspidy @gajasinski

**Comment by merapi** [comment]
We are still waiting for the design pod feedback on this.

**Comment by swissspidy** [comment]
@felipebochehin87 I think this issue exists also outside of this PR:

![Screenshot 2022-06-30 at 16 07 29](https://user-images.githubusercontent.com/841956/176698336-4caf8243-60e5-4513-9a07-ce9653ca0299.png)
 
So it would be something that warrants its own ticket. Though it's a a bit of an edge case, so doesn't seem high prio to me.

**Comment by felipebochehin87** [comment]
Thanks @swissspidy . I'll create a new ticket just so we have a record :)
